File: backend/src/main/java/org/example/backend/model/entities/Contract.java
package org.example.backend.model.entities;

import lombok.*;
import org.springframework.data.annotation.Id;

import java.time.LocalDate;

/**
 * Vertrag
 */
@With
@Builder
public record Contract(
        @Id
        String id,
        String title,
        String description,
        LocalDate startDate,
        LocalDate endDate,
        String extractedText,
        int aiLevel,
        String aiAnalysisText,
        String fileName,
        byte[] file
) {
}

File: frontend/src/model/Contract.ts
export type Contract = {
    id: string;
    title: string;
    description: string;
    startDate: string;     // kommt als dd.MM.yyyy vom Backend
    endDate: string;       // kommt als dd.MM.yyyy vom Backend
    aiLevel: number;
    aiAnalysisText: string;
    fileName: string;
    fileBase64: string | null; // Base64 String vom Backend
};
